woocommerce/woocommerce
# Fix "Coming soon" bar styles are missing in several themes

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/52231

The WooCommerce Coming Soon notice bar is not rendering correctly across multiple WordPress themes. Instead of appearing as a fixed bar at the bottom of the screen, the notice text appears inline at the bottom of the page. This issue affects:
- Storefront
- Twenty Twenty Three
- Twenty Nineteen
- Likely other WP core themes prior to Twenty Twenty Three

The root cause appears to be related to CSS handling:
- WP core themes prior to TT3 use custom `woocommerce.css`
- Storefront [dequeues `woocommerce.css`](https://github.com/woocommerce/storefront/blob/71002541618506d08e5475896a38d9e61d63a9e4/inc/woocommerce/class-storefront-woocommerce.php#L39)

This PR adds a dedicated stylesheet for coming soon banner and ensure styles are loaded regardless of theme's CSS handling.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable Coming Soon mode on a fresh site
2. Test with affected themes:
  - Storefront
  - Twenty Twenty Three
  - Twenty Nineteen
  - Other WP core themes
3. Verify Coming Soon bar appears fixed at bottom of screen
4. Verify `wp-content/plugins/woocommerce/assets/css/coming-soon.css` is not loaded when:
  - Store is live
  - Visiting admin pages
  - Coming Soon mode is enabled but non-admin user is logged in

Before
![Screenshot 2024-10-24 at 15 31 22](https://github.com/user-attachments/assets/b0a62c8b-647e-45c4-a3bf-f3ab0fc52d54)

After
![Screenshot 2024-10-24 at 15 30 56](https://github.com/user-attachments/assets/05042b79-1ed7-4f63-a059-61da4878ee66)




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
